File: typeScript/challenges/challenge25.ts
export function execute (code: string): number {
  let value = 0
  const moves: Record<string, number> = {
    '+': 1,
    '-': -1
  }

  for (let i = 0; i < code.length; i += 1) {
    const move = moves[code[i]]
    if (move !== undefined) {
      value += move
      continue
    }

    if (code[i] === '>') continue
    if (code[i] === '[') {
      value = 0
      i = code.indexOf(']', i)
      continue
    }
    if (value === 0) i = code.indexOf('}', i)
  }

  return value
}

File: python/challenge25.py
def execute(code: str) -> str:
    counter = 0
    aux = 0
    while aux < len(code):
        #print(code[aux])
        if code[aux] == '+':
            counter += 1
        elif code[aux] == '-':
            counter -= 1
        elif code[aux] == '[':
            counter = 0
            while code[aux] != ']':
                aux += 1
        elif code[aux] == '{':
            if counter == 0:
                while code[aux] != '}':
                    aux += 1
        #print(aux)
        aux += 1
    return counter

print(execute('>+++[-]')) #0
print(execute('+--')) #-1
print(execute('+{+}')) #2
print(execute('-[++{-}]+{++++}')) #5